### PR TITLE
Add template for data.kitware.com to ExternalData_URL_TEMPLATES

### PR DESCRIPTION
**Pull request by cdeepakroy**
This will allow the storage of external data used for tests on data.kitware.com

### CODE REVIEW

**Comment by cdeepakroy** [comment]
@zachmullen PTAL

File: tests/TestData.cmake [hunk 1]
```diff
@@ -14,6 +14,7 @@ set(_plugin_build_data_path "${_build_data_path}/plugins")
 # set the default mirror
 set(ExternalData_URL_TEMPLATES
   "https://midas3.kitware.com/midas/api/rest?method=midas.bitstream.download&checksum=%(hash)&algorithm=%(algo)"
+  "https://data.kitware.com:443/api/v1/file/hashsum/%(algo)/%(hash)/download"
```
Comment (zachmullen) [inline tests/TestData.cmake]: Is inclusion of the port necessary?

Comment (cdeepakroy) [inline tests/TestData.cmake]: Ah! Just removed it. Works without it.

**Comment by zachmullen** [comment]
Lgtm

**Pull request merged**